src/Fable.Cli/Main.fs

```diff
@@ -144,11 +144,55 @@ module private Util =
             if fileExt.EndsWith(".ts") then Path.replaceExtension ".js" fileExt else fileExt
         let sourceDir = Path.GetDirectoryName(sourcePath)
         let targetDir = Path.GetDirectoryName(targetPath)
-        let stream = new IO.StreamWriter(targetPath)
+        let memoryStream = new IO.MemoryStream()
+        let stream = new IO.StreamWriter(memoryStream)
         let mapGenerator = lazy (SourceMapSharp.SourceMapGenerator(?sourceRoot = cliArgs.SourceMapsRoot))
+        let WriteStreamToFileAsync(stream : IO.Stream, filePath : string) = async {
+            stream.Seek(0, IO.SeekOrigin.Begin) |> ignore
+            use fileStream = new IO.StreamWriter(filePath)
+            do! stream.CopyToAsync(fileStream.BaseStream) |> Async.AwaitTask
+            do! fileStream.FlushAsync() |> Async.AwaitTask
+        }
+        let IsMemoryStreamEqualToFileAsync() = async {
+            memoryStream.Seek(0, IO.SeekOrigin.Begin) |> ignore
+            use fileStream = IO.File.OpenRead(targetPath)
+
+            let buffer = Array.zeroCreate 1024
+
+            let readBytesAsync (stream: IO.Stream) = async {
+                let rec nextBytesAsync() = async {
+                    let! count = stream.AsyncRead(buffer, 0, buffer.Length)
+                    if count = 0 then return None
+                    else 
+                        let res = 
+                            if count = buffer.Length then buffer
```

Comment:
As we are using a single buffer, if `count = 0` we just return a reference to the buffer. But in the lines below:

```fsharp
                    let! nextBytes1 = readBytesAsync memoryStream
                    let! nextBytes2 = readBytesAsync fileStream 
```

`nextBytes1` is going to be same as `nextBytes2` (if count is 0 in both cases), because `readBytesAsync` returns the same array reference in both cases and the second `readBytesAsync` will overwrite the buffer.